evm/exceptions.py

```diff
@@ -150,3 +150,7 @@ class UnannouncedStateAccess(VMError):
     been specified in the transaction's read or write list, respectively.
     """
     pass
+
+
+class PAYGASResetGasprice(Exception):
```

Comment:
other name suggestions:

- `GasPriceAlreadySet`
- `GasPriceOverwriteProtection`

I'm thinking the inclusing of `PAYGAS` in the name isn't needed and is a little visually confusing.